static/app/views/organizationContext.tsx

```diff
@@ -110,8 +108,7 @@ export function OrganizationContextProvider({children}: Props) {
     }
 
     metric.mark({name: 'organization-details-fetch-start'});
-    fetchOrganizationDetails(api, orgSlug, false, hasMadeFirstFetch.current);
-    hasMadeFirstFetch.current = true;
+    fetchOrganizationDetails(api, orgSlug, false, true);
   }, [api, orgSlug, organization]);
```

Comment:
Did you check what happens if someone changes orgs while staying on the page? The preload only ever fires once while the `index.html` is loading, that's the reason behind it switching the bool to off after first load, since afterward we can't trust it's either up to date or the right org.

Reply:
Switching orgs worked in `dev-ui` as the domain name changes which triggers a new page load. The preloader also takes the org and slug property as parameters. If the slug doesn't match the preloaded data the fallback loader is used instead and fresh data is fetched.